core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -140,23 +145,18 @@ public void sync() throws IOException {
         if (syncNeeded()) {
             synchronized (this) {
                 ensureOpen(); // this call gives a better exception that the incRef if we are closed by a tragic event
-                channelReference.incRef();
+                final long offsetToSync;
+                final int opsCounter;
+                outputStream.flush();
```

Comment:
haha. I changed this while working on other changes, before reaching the comment. Agreed :)  . I'll change the java docs.